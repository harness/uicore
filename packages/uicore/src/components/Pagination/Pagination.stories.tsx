/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import type { Meta, Story } from '@storybook/react'

import Pagination, { PaginationProps } from './Pagination'

export default {
  title: 'Components / Pagination',
  component: Pagination
} as Meta

export const Basic: Story<PaginationProps> = args => {
  const [pageIndex, setPageIndex] = useState(args.pageIndex || 0)

  return <Pagination {...args} pageIndex={pageIndex} gotoPage={setPageIndex} />
}

Basic.args = {
  itemCount: 147,
  pageCount: 15,
  pageSize: 10
}

export const VariablePageSize: Story<PaginationProps> = args => {
  const [pageIndex, setPageIndex] = useState(args.pageIndex || 0)
  const [pageSize, setpageSize] = useState(args.pageSize || 0)

  return (
    <Pagination
      {...args}
      pageSize={pageSize}
      pageIndex={pageIndex}
      gotoPage={setPageIndex}
      pageSizeOptions={args.pageSizeOptions}
      onPageSizeChange={setpageSize}
    />
  )
}

VariablePageSize.args = {
  itemCount: 147,
  pageCount: 15,
  pageSize: 10,
  pageSizeOptions: [5, 10, 25, 50]
}

export const HideNumberButtonsAtBreakpoint: Story<PaginationProps> = args => <Pagination {...args} />

HideNumberButtonsAtBreakpoint.args = {
  hidePageNumbers: false,
  itemCount: 147,
  pageCount: 15,
  pageSize: 10,
  breakAt: 830,
  gotoPage: () => null
}
