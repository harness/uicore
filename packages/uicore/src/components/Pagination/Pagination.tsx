/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import cx from 'classnames'
import { Menu, MenuItem } from '@blueprintjs/core'
import { Select, ItemListRenderer } from '@blueprintjs/select'

import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import { Button, ButtonSize } from '../Button/Button'
import { SelectOption } from '../Select/Select'
import { FontVariation } from '@harness/design-system'
import { DropDown, DropDownProps } from '../DropDown/DropDown'
import useWindowWidth from '../../hooks/useWindowWidth'

import css from './Pagination.css'
import { IconName } from '@harness/icons'
export type PaginationActions = 'Prev' | 'PageNumber' | 'Next'

export interface PaginationProps {
  pageSize: number
  pageCount: number
  itemCount: number
  pageIndex?: number
  gotoPage?: (index: number, context?: PaginationActions) => void
  pageSizeOptions?: number[]
  hidePageNumbers?: boolean
  className?: string
  onPageSizeChange?: (newPageSize: number) => void
  breakAt?: number
  showPagination?: boolean
  pageSizeDropdownProps?: Partial<DropDownProps>
  hidePageSize?: boolean
  minimal?: boolean
  classNames?: {
    pageNumbersClassName?: string
    buttonLeftClassName?: string
    buttonRightClassName?: string
    pageNumbersLayoutClassName?: string
    pageSizeOptionsLayoutClassName?: string
    pageSizeDropdownClassName?: string
    pageSizeFeedbackClassName?: string
    moreButtonClassName?: string
    moreLeftIcon?: IconName
    moreRightIcon?: IconName
  }
}

interface PageNumbersProps {
  pageCount: number
  pageCountClamp: number
  pageIndex: number
  gotoPage: Required<PaginationProps>['gotoPage']
  className?: string
  moreRightIcon?: IconName
  moreLeftIcon?: IconName
  moreButtonClassName?: string
}

const CustomSelect = Select.ofType<SelectOption>()

const renderMenu: ItemListRenderer<SelectOption> = ({ items, itemsParentRef, renderItem }) => {
  const renderedItems = items.map(renderItem).filter(item => item !== null)
  return (
    <Menu ulRef={itemsParentRef} className={css.pageSizeDropdown}>
      {renderedItems}
    </Menu>
  )
}

const PageNumbers: React.FC<PageNumbersProps> = ({
  pageCount,
  pageCountClamp,
  pageIndex,
  gotoPage,
  className,
  moreButtonClassName
}) => {
  const moreLeft = pageIndex + 1 > pageCountClamp
  const moreRight = pageCount - pageIndex > pageCountClamp

  const renderPageButton = (pageNumber: number, buttonClassName = '') => {
    const isCurrent = pageIndex + 1 === pageNumber
    return (
      <Button
        key={`page-${pageNumber}`}
        text={pageNumber}
        disabled={isCurrent}
        onClick={() => gotoPage(pageNumber - 1, 'PageNumber')}
        className={cx(className, buttonClassName, css.roundedButton, { [css.selected]: isCurrent })}
      />
    )
  }

  return (
    <>
      {pageCount <= pageCountClamp * 2 ? (
        Array.from(Array(pageCount).keys()).map(pageNumber => {
          const isCurrent = pageIndex === pageNumber
          return (
            <Button
              key={`page-${pageNumber}`}
              text={pageNumber + 1}
              disabled={isCurrent}
              className={cx(className, css.roundedButton, { [css.selected]: isCurrent })}
              onClick={() => gotoPage(pageNumber, 'PageNumber')}
            />
          )
        })
      ) : (
        <React.Fragment>
          {Array.from({ length: moreLeft ? 2 : pageCountClamp }, (_, i) => renderPageButton(i + 1))}
          {moreLeft && (
            <CustomSelect
              className={moreButtonClassName}
              filterable={false}
              items={Array.from(Array(moreRight ? pageIndex - 3 : pageCount - pageCountClamp - 2).keys()).map(
                pageNumber => {
                  return {
                    label: `${pageNumber + 3}`,
                    value: pageNumber + 2
                  }
                }
              )}
              itemListRenderer={renderMenu}
              itemRenderer={(item, { handleClick }) => (
                <MenuItem key={item.label} text={item.label} onClick={handleClick} />
              )}
              onItemSelect={item => gotoPage(item.value as number, 'PageNumber')}>
              <Button icon={'more'} className={cx(css.roundedButton, moreButtonClassName)} />
            </CustomSelect>
          )}
          {moreLeft && moreRight && Array.from({ length: 3 }, (_, i) => renderPageButton(pageIndex + i))}
          {moreRight && (
            <CustomSelect
              className={moreButtonClassName}
              filterable={false}
              items={Array.from(
                Array(moreLeft ? pageCount - pageIndex - 4 : pageCount - pageCountClamp - 2).keys()
              ).map(pageNumber => {
                return {
                  label: `${pageNumber + (moreLeft ? pageIndex + 3 : pageCountClamp + 1)}`,
                  value: pageNumber + (moreLeft ? pageIndex + 2 : pageCountClamp)
                }
              })}
              itemListRenderer={renderMenu}
              itemRenderer={(item, { handleClick }) => (
                <MenuItem key={item.label} text={item.label} onClick={handleClick} className={css.paginationMenu} />
              )}
              onItemSelect={item => gotoPage(item.value as number, 'PageNumber')}>
              <Button icon={'more'} className={cx(css.roundedButton, moreButtonClassName)} />
            </CustomSelect>
          )}
          {moreRight
            ? [pageCount - 1, pageCount].map(i => renderPageButton(i))
            : Array.from({ length: pageCountClamp }, (_, i) => renderPageButton(i + 1 + pageCount - pageCountClamp))}
        </React.Fragment>
      )}
    </>
  )
}

const Pagination: React.FC<PaginationProps> = props => {
  const {
    pageSize,
    pageIndex = 0,
    gotoPage,
    pageCount,
    itemCount,
    pageSizeOptions,
    onPageSizeChange,
    hidePageNumbers,
    className,
    breakAt,
    showPagination,
    pageSizeDropdownProps,
    hidePageSize = false,
    minimal = false,
    classNames
  } = props
  const currentWindowWidth = useWindowWidth()
  const {
    pageNumbersClassName,
    buttonLeftClassName,
    buttonRightClassName,
    pageNumbersLayoutClassName,
    pageSizeOptionsLayoutClassName,
    pageSizeDropdownClassName,
    pageSizeFeedbackClassName,
    moreButtonClassName,
    moreLeftIcon,
    moreRightIcon
  } = classNames || {}
  const selectOptions: SelectOption[] =
    pageSizeOptions?.map(option => {
      return { label: option.toString(), value: option.toString() }
    }) || []

  if (pageIndex >= pageCount) {
    // eslint-disable-next-line no-console
    console.warn(`Pagination: pageIndex (${pageIndex}) can't be more than/equal to pageCount (${pageCount})`)
    return null
  }
  if (pageSize * pageCount < itemCount) {
    // eslint-disable-next-line no-console
    console.warn(
      `Pagination: pageSize (${pageSize}) * pageCount (${pageCount}) can't be less than itemCount (${itemCount})`
    )
  }

  const showNumbers = !hidePageNumbers && (!breakAt || breakAt <= currentWindowWidth)
  const showPaginationComponent = !!showPagination || itemCount > pageSize

  return showPaginationComponent ? (
    <Layout.Horizontal
      padding={{ top: 'medium', bottom: 'medium' }}
      flex={{ align: 'center-center', distribution: 'space-between' }}
      className={cx(className, css.container)}>
      {itemCount <= pageSize ? (
        // eg. 3 of 3
        <Text inline font={{ variation: FontVariation.SMALL_SEMI }}>
          {itemCount} of {itemCount}
        </Text>
      ) : (
        // eg. (1-10) of 50
        <Text inline font={{ variation: FontVariation.SMALL_SEMI }}>
          {`(${pageSize * pageIndex + 1} - ${Math.min(pageSize * (pageIndex + 1), itemCount)}) of ${itemCount}`}
        </Text>
      )}
      <Layout.Horizontal className={pageNumbersLayoutClassName}>
        <Button
          text={!minimal && 'Prev'}
          icon={moreLeftIcon ? moreLeftIcon : 'arrow-left'}
          size={ButtonSize.SMALL}
          className={cx(css.roundedButton, css.buttonLeft, buttonLeftClassName)}
          iconProps={{ size: 12 }}
          onClick={() => gotoPage?.(pageIndex - 1, 'Prev')}
          disabled={pageIndex === 0}
        />
        {gotoPage && showNumbers ? (
          <PageNumbers
            gotoPage={gotoPage}
            pageIndex={pageIndex}
            pageCountClamp={5}
            pageCount={pageCount}
            className={pageNumbersClassName}
            moreButtonClassName={moreButtonClassName}
          />
        ) : null}
        <Button
          text={!minimal && 'Next'}
          rightIcon={moreRightIcon ? moreRightIcon : 'arrow-right'}
          size={ButtonSize.SMALL}
          className={cx(css.roundedButton, css.buttonRight, buttonRightClassName)}
          iconProps={{ size: 12 }}
          onClick={() => gotoPage?.(pageIndex + 1, 'Next')}
          disabled={pageIndex === pageCount - 1}
        />
      </Layout.Horizontal>
      {!hidePageSize &&
        (pageSizeOptions && onPageSizeChange ? (
          <Layout.Horizontal
            flex={{ alignItems: 'baseline' }}
            spacing="small"
            className={pageSizeOptionsLayoutClassName}>
            <Text className="show">Show</Text>
            <DropDown
              items={selectOptions}
              onChange={item => onPageSizeChange(item.value as number)}
              value={pageSize.toString()}
              filterable={false}
              width={60}
              className={cx(css.pageSizeDropdown, pageSizeDropdownClassName)}
              {...pageSizeDropdownProps}
            />
            <Text className="per-page">per page</Text>
          </Layout.Horizontal>
        ) : (
          <span className={pageSizeFeedbackClassName}>{`Showing ${pageSize} per page`}</span>
        ))}
    </Layout.Horizontal>
  ) : null
}

export default Pagination
