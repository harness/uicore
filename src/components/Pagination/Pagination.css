.roundedButton {
  height: 32px !important;
  background-color: var(--white) !important;
  box-shadow: 0px 0px 1px rgba(40, 41, 61, 0.04), 0px 2px 4px rgba(96, 97, 112, 0.16) !important;
  border: none !important;
  opacity: 1 !important;
  border-radius: 0 !important;
  outline: none !important;

  &:hover {
    background-color: var(--primary-1) !important;
  }

  &:disabled {
    color: var(--grey-400) !important;
    background-color: var(--white) !important;
  }

  &.selected {
    background-color: var(--primary-6) !important;
    color: var(--white) !important;
  }
}

.buttonLeft {
  border-radius: 32px 0 0 32px !important;
}

.buttonRight {
  border-radius: 0 32px 32px 0 !important;
}

.pageSizeDropdown {
  max-height: 200px;
  overflow-y: auto;
  min-width: 60px;
}

.container {
  gap: var(--spacing-small);
}
