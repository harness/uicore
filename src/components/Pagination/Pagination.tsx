/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import cx from 'classnames'
import { Menu, MenuItem } from '@blueprintjs/core'
import { Select, ItemListRenderer } from '@blueprintjs/select'

import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import { Button, ButtonSize } from '../Button/Button'
import { SelectOption } from '../Select/Select'
import { FontVariation } from '@harness/design-system'
import { DropDown } from '../DropDown/DropDown'
import useWindowWidth from '../../hooks/useWindowWidth'

import css from './Pagination.css'

export interface PaginationProps {
  pageSize: number
  pageCount: number
  itemCount: number
  pageIndex?: number
  gotoPage?: (index: number) => void
  pageSizeOptions?: number[]
  hidePageNumbers?: boolean
  className?: string
  onPageSizeChange?: (newPageSize: number) => void
  breakAt?: number
}

interface PageNumbersProps {
  pageCount: number
  pageCountClamp: number
  pageIndex: number
  gotoPage: (index: number) => void
}

const CustomSelect = Select.ofType<SelectOption>()

const renderMenu: ItemListRenderer<SelectOption> = ({ items, itemsParentRef, renderItem }) => {
  const renderedItems = items.map(renderItem).filter(item => item !== null)
  return (
    <Menu ulRef={itemsParentRef} className={css.pageSizeDropdown}>
      {renderedItems}
    </Menu>
  )
}

const PageNumbers: React.FC<PageNumbersProps> = ({ pageCount, pageCountClamp, pageIndex, gotoPage }) => {
  const moreLeft = pageIndex + 1 > pageCountClamp
  const moreRight = pageCount - pageIndex > pageCountClamp

  const renderPageButton = (pageNumber: number) => {
    const isCurrent = pageIndex + 1 === pageNumber
    return (
      <Button
        key={`page-${pageNumber}`}
        text={pageNumber}
        disabled={isCurrent}
        onClick={() => gotoPage(pageNumber - 1)}
        className={cx(css.roundedButton, { [css.selected]: isCurrent })}
      />
    )
  }

  return (
    <>
      {pageCount <= pageCountClamp * 2 ? (
        Array.from(Array(pageCount).keys()).map(pageNumber => {
          const isCurrent = pageIndex === pageNumber
          return (
            <Button
              key={`page-${pageNumber}`}
              text={pageNumber + 1}
              disabled={isCurrent}
              className={cx(css.roundedButton, { [css.selected]: isCurrent })}
              onClick={() => gotoPage(pageNumber)}
            />
          )
        })
      ) : (
        <React.Fragment>
          {Array.from({ length: moreLeft ? 2 : pageCountClamp }, (_, i) => renderPageButton(i + 1))}
          {moreLeft && (
            <CustomSelect
              filterable={false}
              items={Array.from(Array(moreRight ? pageIndex - 3 : pageCount - pageCountClamp - 2).keys()).map(
                pageNumber => {
                  return {
                    label: `${pageNumber + 3}`,
                    value: pageNumber + 2
                  }
                }
              )}
              itemListRenderer={renderMenu}
              itemRenderer={(item, { handleClick }) => (
                <MenuItem key={item.label} text={item.label} onClick={handleClick} />
              )}
              onItemSelect={item => gotoPage(item.value as number)}>
              <Button icon="more" className={css.roundedButton} />
            </CustomSelect>
          )}
          {moreLeft && moreRight && Array.from({ length: 3 }, (_, i) => renderPageButton(pageIndex + i))}
          {moreRight && (
            <CustomSelect
              filterable={false}
              items={Array.from(
                Array(moreLeft ? pageCount - pageIndex - 4 : pageCount - pageCountClamp - 2).keys()
              ).map(pageNumber => {
                return {
                  label: `${pageNumber + (moreLeft ? pageIndex + 3 : pageCountClamp + 1)}`,
                  value: pageNumber + (moreLeft ? pageIndex + 2 : pageCountClamp)
                }
              })}
              itemListRenderer={renderMenu}
              itemRenderer={(item, { handleClick }) => (
                <MenuItem key={item.label} text={item.label} onClick={handleClick} />
              )}
              onItemSelect={item => gotoPage(item.value as number)}>
              <Button icon="more" className={css.roundedButton} />
            </CustomSelect>
          )}
          {moreRight
            ? [pageCount - 1, pageCount].map(i => renderPageButton(i))
            : Array.from({ length: pageCountClamp }, (_, i) => renderPageButton(i + 1 + pageCount - pageCountClamp))}
        </React.Fragment>
      )}
    </>
  )
}

const Pagination: React.FC<PaginationProps> = props => {
  const {
    pageSize,
    pageIndex = 0,
    gotoPage,
    pageCount,
    itemCount,
    pageSizeOptions,
    onPageSizeChange,
    hidePageNumbers,
    className,
    breakAt
  } = props
  const currentWindowWidth = useWindowWidth()

  const selectOptions: SelectOption[] =
    pageSizeOptions?.map(option => {
      return { label: option.toString(), value: option.toString() }
    }) || []

  if (pageIndex >= pageCount) {
    // eslint-disable-next-line no-console
    console.warn(`Pagination: pageIndex (${pageIndex}) can't be more than/equal to pageCount (${pageCount})`)
    return null
  }
  if (pageSize * pageCount < itemCount) {
    // eslint-disable-next-line no-console
    console.warn(
      `Pagination: pageSize (${pageSize}) * pageCount (${pageCount}) can't be less than itemCount (${itemCount})`
    )
  }

  const showNumbers = !hidePageNumbers && (!breakAt || breakAt <= currentWindowWidth)

  return (
    <Layout.Horizontal
      padding={{ top: 'medium', bottom: 'medium' }}
      flex={{ align: 'center-center', distribution: 'space-between' }}
      className={cx(className, css.container)}>
      {itemCount <= pageSize ? (
        // eg. 3 of 3
        <Text inline font={{ variation: FontVariation.SMALL_SEMI }}>
          {itemCount} of {itemCount}
        </Text>
      ) : (
        // eg. (1-10) of 50
        <Text inline font={{ variation: FontVariation.SMALL_SEMI }}>
          {`(${pageSize * pageIndex + 1} - ${Math.min(pageSize * (pageIndex + 1), itemCount)}) of ${itemCount}`}
        </Text>
      )}
      <Layout.Horizontal>
        <Button
          text="Prev"
          icon="arrow-left"
          size={ButtonSize.SMALL}
          className={cx(css.roundedButton, css.buttonLeft)}
          iconProps={{ size: 12 }}
          onClick={() => gotoPage?.(pageIndex - 1)}
          disabled={pageIndex === 0}
        />
        {gotoPage && showNumbers ? (
          <PageNumbers gotoPage={gotoPage} pageIndex={pageIndex} pageCountClamp={5} pageCount={pageCount} />
        ) : null}
        <Button
          text="Next"
          rightIcon="arrow-right"
          size={ButtonSize.SMALL}
          className={cx(css.roundedButton, css.buttonRight)}
          iconProps={{ size: 12 }}
          onClick={() => gotoPage?.(pageIndex + 1)}
          disabled={pageIndex === pageCount - 1}
        />
      </Layout.Horizontal>

      {pageSizeOptions && onPageSizeChange ? (
        <Layout.Horizontal flex={{ alignItems: 'baseline' }} spacing="small">
          <Text>Show</Text>
          <DropDown
            items={selectOptions}
            onChange={item => onPageSizeChange(item.value as number)}
            value={pageSize.toString()}
            filterable={false}
            width={60}
            className={css.pageSizeDropdown}
          />
          <Text>per page</Text>
        </Layout.Horizontal>
      ) : (
        <span>{`Showing ${pageSize} per page`}</span>
      )}
    </Layout.Horizontal>
  )
}

export default Pagination
