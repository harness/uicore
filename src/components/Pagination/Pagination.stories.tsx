import React, { useState } from 'react'
import type { Meta, Story } from '@storybook/react'

import Pagination, { PaginationProps } from './Pagination'

export default {
  title: 'Components / Pagination',
  component: Pagination
} as Meta

export const Basic: Story<PaginationProps> = args => {
  const [pageIndex, setPageIndex] = useState(args.pageIndex || 0)

  return <Pagination {...args} pageIndex={pageIndex} gotoPage={setPageIndex} />
}

Basic.args = {
  itemCount: 97,
  pageCount: 10,
  pageSize: 10
}
