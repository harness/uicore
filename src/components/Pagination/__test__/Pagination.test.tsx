import React from 'react'
import { render } from '@testing-library/react'
import Pagination from '../Pagination'

const noop = () => {
  // empty function
}

describe('Pagination', () => {
  test('itemCount < pageSize', () => {
    const { container } = render(<Pagination pageSize={10} pageIndex={0} pageCount={1} itemCount={5} gotoPage={noop} />)
    expect(container).toMatchSnapshot()
  })

  test('pageCount < 8', () => {
    const { container } = render(
      <Pagination pageSize={10} pageIndex={0} pageCount={3} itemCount={30} gotoPage={noop} />
    )
    expect(container).toMatchSnapshot()
  })

  test('pageCount >= 8', () => {
    const { container } = render(
      <Pagination pageSize={10} pageIndex={0} pageCount={10} itemCount={100} gotoPage={noop} />
    )
    expect(container).toMatchSnapshot()
  })
})
