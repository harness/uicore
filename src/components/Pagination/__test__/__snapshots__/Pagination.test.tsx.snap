// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination itemCount < pageSize 1`] = `
<div>
  <div
    class="horizontal layout-spacing-undefined main container padding padding-top padding-top-medium padding-bottom padding-bottom-medium flex flex-align flex-align-center-center flex-distribution flex-distribution-space-between"
  >
    <span
      class="font main inline inline-true font-variation font-variation-small-semi"
    >
      5
       of 
      5
    </span>
    <div
      class="horizontal layout-spacing-undefined main"
    >
      <button
        class="bp3-button bp3-disabled button font main roundedButton buttonLeft with-current-color withLeftIcon size small"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-arrow-left main padding padding-right padding-right-xsmall"
          icon="arrow-left"
        >
          <svg
            data-icon="arrow-left"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-left
            </desc>
            <path
              d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          Prev
        </span>
      </button>
      <button
        class="bp3-button bp3-disabled button font main roundedButton selected with-current-color"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          1
        </span>
      </button>
      <button
        class="bp3-button bp3-disabled button font main roundedButton buttonRight with-current-color withRightIcon size small"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Next
        </span>
        <span
          class="bp3-icon bp3-icon-arrow-right main"
          icon="arrow-right"
        >
          <svg
            data-icon="arrow-right"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-right
            </desc>
            <path
              d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
    <span>
      Showing 10 per page
    </span>
  </div>
</div>
`;

exports[`Pagination pageCount < 8 1`] = `
<div>
  <div
    class="horizontal layout-spacing-undefined main container padding padding-top padding-top-medium padding-bottom padding-bottom-medium flex flex-align flex-align-center-center flex-distribution flex-distribution-space-between"
  >
    <span
      class="font main inline inline-true font-variation font-variation-small-semi"
    >
      (1 - 10) of 30
    </span>
    <div
      class="horizontal layout-spacing-undefined main"
    >
      <button
        class="bp3-button bp3-disabled button font main roundedButton buttonLeft with-current-color withLeftIcon size small"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-arrow-left main padding padding-right padding-right-xsmall"
          icon="arrow-left"
        >
          <svg
            data-icon="arrow-left"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-left
            </desc>
            <path
              d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          Prev
        </span>
      </button>
      <button
        class="bp3-button bp3-disabled button font main roundedButton selected with-current-color"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          1
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          2
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          3
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton buttonRight with-current-color withRightIcon size small"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Next
        </span>
        <span
          class="bp3-icon bp3-icon-arrow-right main"
          icon="arrow-right"
        >
          <svg
            data-icon="arrow-right"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-right
            </desc>
            <path
              d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
    <span>
      Showing 10 per page
    </span>
  </div>
</div>
`;

exports[`Pagination pageCount >= 8 1`] = `
<div>
  <div
    class="horizontal layout-spacing-undefined main container padding padding-top padding-top-medium padding-bottom padding-bottom-medium flex flex-align flex-align-center-center flex-distribution flex-distribution-space-between"
  >
    <span
      class="font main inline inline-true font-variation font-variation-small-semi"
    >
      (1 - 10) of 100
    </span>
    <div
      class="horizontal layout-spacing-undefined main"
    >
      <button
        class="bp3-button bp3-disabled button font main roundedButton buttonLeft with-current-color withLeftIcon size small"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-arrow-left main padding padding-right padding-right-xsmall"
          icon="arrow-left"
        >
          <svg
            data-icon="arrow-left"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-left
            </desc>
            <path
              d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          Prev
        </span>
      </button>
      <button
        class="bp3-button bp3-disabled button font main roundedButton selected with-current-color"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          1
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          2
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          3
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          4
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          5
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          6
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          7
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          8
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          9
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton with-current-color"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          10
        </span>
      </button>
      <button
        class="bp3-button button font main roundedButton buttonRight with-current-color withRightIcon size small"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Next
        </span>
        <span
          class="bp3-icon bp3-icon-arrow-right main"
          icon="arrow-right"
        >
          <svg
            data-icon="arrow-right"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <desc>
              arrow-right
            </desc>
            <path
              d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
    <span>
      Showing 10 per page
    </span>
  </div>
</div>
`;
